sql/catalyst/src/main/scala/org/apache/spark/sql/internal/StaticSQLConf.scala

```diff
@@ -170,6 +170,16 @@ object StaticSQLConf {
       .intConf
       .createWithDefault(1000)
 
+  val SHUFFLE_EXCHANGE_MAX_THREAD_THRESHOLD =
+    buildStaticConf("spark.sql.shuffleExchange.maxThreadThreshold")
+      .internal()
+      .doc("The maximum degree of parallelism for doing preparation of shuffle exchange, " +
+        "which includes subquery execution, file listing, etc.")
+      .version("4.0.0")
+      .intConf
+      .checkValue(thres => thres > 0 && thres <= 1024, "The threshold must be in (0,1024].")
+      .createWithDefault(1024)
```

Comment:
Can you explain why we have this number? It might create memory pressure on the driver

Reply:
The shuffle async job is just waiting for other work (subquery expression execution) to finish, which is very light-weighted. The broadcast async job executes a query and collects the result in the driver, which is very heavy. That's why we can give much larger parallelism to the shuffle async jobs. In our benchmark we found this number is reasonably good for TPC.

Reply:
Is there a correlation with the number of system cores?

Reply:
I don't think so, the `BROADCAST_EXCHANGE_MAX_THREAD_THRESHOLD` is also way larger than the driver system cores.

Reply:
I'm not sure if this parameter has anything to do with SPARK-49091 or if it was just caused by SPARK-41914 which the reporter pointed to.

Also cc @wangyum

Reply:
Update: SPARK-49091 is not related